packages/kolibri-common/components/SidePanelModal/index.vue

```diff
@@ -0,0 +1,272 @@
+<template>
+
+  <div
+    ref="sidePanel"
+    :tabindex="0"
+    :class="{ 'is-rtl': isRtl, 'is-mobile': isMobile }"
+    @keyup.esc="closePanel"
+  >
+    <transition name="side-panel">
+      <FocusTrap
+        @shouldFocusFirstEl="$emit('shouldFocusFirstEl')"
+        @shouldFocusLastEl="focusLastEl"
+      >
+        <section
+          class="side-panel"
+          role="region"
+          :style="sidePanelStyles"
+          :aria-label="ariaLabel"
+        >
+
+          <!-- Fixed header -->
+          <div
+            v-show="$slots.header"
+            ref="fixedHeader"
+            class="fixed-header"
+            :style="fixedHeaderStyles"
+          >
+            <div class="header-content">
+              <slot name="header">
+              </slot>
+            </div>
+          </div>
+
+          <KIconButton
+            v-if="closeButtonIconType"
+            :icon="closeButtonIconType"
+            class="close-button"
+            :style="closeButtonStyle"
+            :ariaLabel="coreString('closeAction')"
+            :tooltip="coreString('closeAction')"
+            @click="closePanel"
+          />
+
+          <!-- Default slot for inserting content which will scroll on overflow -->
+          <div class="side-panel-content" :style="contentStyles">
+            <slot></slot>
+          </div>
+
+        </section>
+      </FocusTrap>
+    </transition>
+
+    <Backdrop
+      :transitions="true"
+      class="backdrop"
+      @click="closePanel"
+    />
+  </div>
+
+</template>
+
+
+<script>
+
+  import Backdrop from 'kolibri.coreVue.components.Backdrop';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import responsiveWindowMixin from 'kolibri.coreVue.mixins.responsiveWindowMixin';
```

Comment:
Just wondering why we are still using this mixin when we have a better alternative in `useKResponsiveWindow` thats more flexible and offers more readability 😄

Reply:
Yes, that does seem like a good thing to switch at this point.